### ISSUE TITLE
Suggested actions does not work in MS Teams.

### ISSUE THREAD

**Issue by ms-86** [OfficeDev/Microsoft-Teams-Samples#1465]
I have checked [NodeJS suggested actions example](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-suggested-actions/nodejs) using Teams Toolkit extension after cloning samples repo on my laptop. Bot sends a welcome message and asks for a favorite color but suggested actions are not displayed as shown on a gif https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-suggested-actions/nodejs/Images/BotSuggestedActionsGif.gif

Then I've download Teams App manifest https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-suggested-actions/csharp/demo-manifest/bot-suggested-actions.zip and uploaded to my M365 developer tenant - behavior is exactly the same, suggested actions are not displayed. MS Docs is now inconsistent when it comes to this particular feature.
MS Teams Developer documentation suggests this feature is supported ([here](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/conversation-messages?tabs=python1%2Capp-manifest-v112-or-later%2Cdotnet2%2Cdotnet3%2Cdotnet4%2Cdotnet5%2Cdesktop%2Cpython#send-suggested-actions) and [here](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/prompt-suggestions?tabs=developer-portal%2Cdesktop%2Cdotnet#suggested-actions-1)) but when you check [Azure Bot Service docs](https://learn.microsoft.com/en-us/azure/bot-service/bot-service-channels-reference?view=azure-bot-service-4.0#card-action-support-by-channel) you will see a None in a Suggested actions column for Microsoft Teams channel.

I have made a Python app as well which I run and test locally with Bot Framework Emulator, and what's interesting those suggested actions are displayed correctly there. After uploading the same app into Azure App Service with accompanying Bot Service and then verifying this in Bot Service Web chat those suggested actions are also displayed - the only issue is with Microsoft Teams.

![image](https://github.com/user-attachments/assets/fcfd56cb-0021-44b6-9653-22f708468906)

**Comment by Prasad-MSFT** [comment]
Thanks for raising this issue. We already have an ICM raised for this, and engineering team is looking into it. 
We will inform you once we get any further update from engineering team. Thanks!

**Comment by Prasad-MSFT** [comment]
@ms-86, according to latest update from engineering team, the fix has been merged. So please do verify at your end and let us know if you still encounter the issue.

**Comment by ms-86** [comment]
I will check this in a couple of days.

**Comment by ms-86** [comment]
I had some time to verify this. Now I can see cards when I chat through MS Teams but I have to admit behaviour is a little bit different and not something I expected to see after doing some tests with Bot Framework Emulator and Test in Web Chat, a feature of the Azure Bot service. Let me explain.

Here's my code for sending an activity with suggested actions:
```python
from botbuilder.core import ActivityHandler, BotAdapter, MessageFactory, TurnContext
from botbuilder.schema import (
    ActionTypes,
    Activity,
    ActivityTypes,
    CardAction,
    ChannelAccount,
    Entity,
    InputHints,
    SuggestedActions,
)
class AiContextBot(ActivityHandler):
    async def on_message_activity(self, turn_context: TurnContext) -> None:
        """
        Handles incoming message activities, processes the message through the chain,
        and sends the response back to the user.

        Args:
            turn_context (TurnContext): The context for the current turn of conversation.
        """
        await turn_context.send_activity(
            Activity(
                type=ActivityTypes.message,
                text="What's your favorite color?",
                input_hint=InputHints.expecting_input,
                suggested_actions=SuggestedActions(
                    actions=[
                        CardAction(
                            type=ActionTypes.im_back,
                            title="RED",
                            value="r",
                        ),
                        CardAction(
                            type=ActionTypes.im_back,
                            title="GREEN",
                            value="g",
                        ),
                        CardAction(
                            type=ActionTypes.im_back,
                            title="BLUE",
                            value="b",
                        ),
                    ],
                    to=[turn_context.activity.from_property.id],
                ),
            )
        )
        return
```
You might have noticed that each CardAction has different values for _type_ and _value_ arguments and this is by design. When I run that code locally and test with Bot Framework Emulator, I can see that after pressing a button of the suggested action, a displayed text (**_title_**) is converted (**_value_**), the same thing with Test in Web Chat functionality of the Azure Bot service. But for some reason the same Bot behaves in a different way when talking through MS Teams - a value that is returned after user presses a button is always equal to **_title_**

**Bot Framework Emulator**
![Image](https://github.com/user-attachments/assets/4d0b1d7a-5c79-49f3-99a5-408b40f5e27d)

**Test in Web Chat (Azure Bot)**
![Image](https://github.com/user-attachments/assets/c76b589d-f651-4bcf-97e2-89639e0b8bd2)

**MS Teams**
![Image](https://github.com/user-attachments/assets/391943de-f74e-450c-b59b-52f56fe75f9f)

**Comment by Prasad-MSFT** [comment]
Yes @ms-86, your observation is absolutely correct and thanks for noticing this. We also observed the same and we have reached out to engineering team to confirm whether it is by design behaviour or a bug. Once we have any information, we will update in this thread.

**Comment by Prasad-MSFT** [comment]
Hi @ms-86 , _quick update:_ we have raised another bug for these 2 issues. Once we get any update from engineering team, we will share in this thread. Thanks!

**Comment by Prasad-MSFT** [comment]
Hi @ms-86, according to latest update from engineering team, this issue is fixed. Please do verify and confirm.
Thanks!

**Comment by lilyydu** [comment]
Closing due to inactivity.
